library/libnames.mli

```diff
@@ -65,23 +65,28 @@ val restrict_path : int -> full_path -> full_path
     qualifications of absolute names, including single identifiers.
     The [qualid] are used to access the name table. *)
 
-type qualid
+type qualid_r
+type qualid = qualid_r CAst.t
```

Comment:
Why do we now expose this?

Reply:
We don't expose more, `qualid_r` is the old `qualid` and the new `qualid` was used in ad-hoc ways in many places.